optimum/intel/openvino/modeling.py

```diff
@@ -952,3 +954,45 @@ def forward(self, **kwargs):
                 model_outputs[key_name] = torch.from_numpy(value).to(self.device) if not np_inputs else value
 
         return ModelOutput(**model_outputs)
+
+
+class OVModelForZeroShotImageClassification(OVModel):
+    auto_model_class = AutoModelForZeroShotImageClassification
+    export_feature = "zero-shot-image-classification"
+
+    def forward(self, input_ids, pixel_values, attention_mask: Optional[torch.Tensor] = None, **kwargs):
+        self.compile()
+
+        np_inputs = isinstance(input_ids, np.ndarray)
+        if not np_inputs:
+            input_ids = input_ids.cpu().numpy()
+            pixel_values = pixel_values.cpu().numpy()
+            attention_mask = attention_mask.cpu().numpy() if attention_mask is not None else attention_mask
+        inputs = {"input_ids": input_ids, "pixel_values": pixel_values}
+        # Add the attention_mask when needed
+        if "attention_mask" in self.input_names:
+            inputs["attention_mask"] = attention_mask
```

Comment:
here shouldn't we check that `attention_mask` is given before inference when needed (and raise an error if None or create a default one) ? not specific to this PR as it's following the same behavior as for other OVModels so no need to include anything just wanted to raise the point

Reply:
added